src/Servers/Kestrel/Core/src/Internal/Http3/QPackHeaderWriter.cs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net.Http.HPack;
+using System.Net.Http.QPack;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http3
+{
+    internal static class QPackHeaderWriter
+    {
+        public static bool BeginEncode(IEnumerator<KeyValuePair<string, string>> enumerator, Span<byte> buffer, out int length)
+        {
+            bool hasValue = enumerator.MoveNext();
+            Debug.Assert(hasValue == true);
```

Comment:
```suggestion
            Debug.Assert(hasValue);
```
It's already true or false.
Should it be `var` instead of `bool` here? Or keep it in the way of dotnet/runtime?

(same below)

Reply:
This is copy and paste from runtime. I'll make this change in a follow-up PR.